common/src/androidTest/java/com/microsoft/identity/common/MsalOAuth2TokenCacheTest.java

```diff
@@ -294,6 +297,48 @@ public void saveTokens() throws Exception {
         assertEquals(defaultTestBundleV2.mGeneratedIdToken, ids.get(0));
     }
 
+    @Test
+    public void saveTokensWithMalformedDataInCache() throws Exception {
+        // Prepopulate the cache with unparseable, junk data
+        mSharedPreferencesFileManager.putString(JUNK_KEY, JUNK_VALUE);
+
+        mOauth2TokenCache.save(
+                mockStrategy,
+                mockRequest,
+                mockResponse
+        );
+
+        final List<AccountRecord> accounts = accountCredentialCache.getAccounts();
+        assertEquals(1, accounts.size());
+        assertEquals(defaultTestBundleV2.mGeneratedAccount, accounts.get(0));
+
+        final List<Credential> credentials = accountCredentialCache.getCredentials();
+        assertEquals(3, credentials.size());
+
+        final List<Credential> rts = new ArrayList<>();
+        final List<Credential> ats = new ArrayList<>();
+        final List<Credential> ids = new ArrayList<>();
+
+        for (final Credential credential : credentials) {
+            if (credential.getCredentialType().equalsIgnoreCase(CredentialType.AccessToken.name())) {
+                ats.add(credential);
+            } else if (credential.getCredentialType().equalsIgnoreCase(CredentialType.RefreshToken.name())) {
+                rts.add(credential);
+            } else if (credential.getCredentialType().equalsIgnoreCase(CredentialType.IdToken.name())) {
+                ids.add(credential);
+            } else {
+                fail();
```

Comment:
nit: if CredentialType is an enum, then you could replace with a switch.  Also, I think you can provide a string in the fail() method saying why it is failing and assist the poor person viewing the test failure to fix it.

Reply:
Applied changes in 8d37efeb340eb3cfe438317e799fa5b3d4320cdb, c6044e92dab13397356bce8bdd4a64ccd673704a